src/pool/srv_pool_scrub_ult.c

```diff
@@ -0,0 +1,358 @@
+/*
+ * (C) Copyright 2021-2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#define D_LOGFAC	DD_FAC(csum)
+
+#include <daos_srv/vos.h>
+#include <daos_srv/srv_csum.h>
+#include <gurt/telemetry_producer.h>
+#include <daos/pool.h>
+#include <daos_prop.h>
+#include "srv_internal.h"
+
+#define C_TRACE(...) D_DEBUG(DB_CSUM, __VA_ARGS__)
+
+/*
+ * telemetry metrics for the scrubber
+ * The target must be parsed out by
+ * lib/telemetry/promexp/collector.go:extractLabels(). If the format here
+ * changes, extractLabels() must be updated as well.
+ */
+#define DF_POOL_DIR "%s/tgt_%d/scrubber"
+#define DP_POOL_DIR(ctx) (ctx)->sc_pool->sp_path, (ctx)->sc_dmi->dmi_tgt_id
+
+#define M_CSUM_COUNTER "csums/current"
+#define M_CSUM_COUNTER_TOTAL "csums/total"
+#define M_CSUM_COUNTER_PREV "csums/prev"
+#define M_BYTES_SCRUBBED "bytes_scrubbed/current"
+#define M_BYTES_SCRUBBED_TOTAL "bytes_scrubbed/total"
+#define M_BYTES_SCRUBBED_PREV "bytes_scrubbed/prev"
+#define M_CSUM_CORRUPTION "corruption/current"
+#define M_CSUM_CORRUPTION_TOTAL "corruption/total"
+#define M_STARTED "scrubber_started"
+#define M_ENDED "scrubber_finished"
+#define M_LAST_DURATION "last_duration"
+
+/*
+ * DAOS_CSUM_SCRUB_DISABLED can be set in the server config to disable the
+ * scrubbing ULT completely for the engine.
+ */
+static inline bool
+scrubbing_is_enabled()
+{
+	char *disabled = getenv("DAOS_CSUM_SCRUB_DISABLED");
```

Comment:
Don't we have a get_env_bool function for this, that would allow true/false values and also log anything that was set.